integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/BeanResource.java

```diff
@@ -164,4 +169,18 @@ String awaitFirst(String key) {
         return list.get(0);
     }
 
+    @Path("/withDefaultBeanCount")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public int withDefaultBeanCount() {
+        return camelContext.getRegistry().findByType(Producers.WithDefaultBeanInstance.class).size();
+    }
+
+    @Path("/withAlternativeBeanCount")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public int withAlternativeBeanCount() {
+        return camelContext.getRegistry().findByType(Producers.WithAlternateBeanInstance.class).size();
+    }
```

Comment:
I see that this primarily covers the cause of the original bug report. But wouldn't it be better to make sure that the registry returns the right single bean returned by the right producer method?

Reply:
That makes sense. thanks for the point, I'll change it/

Reply:
- I fixed tests to control also bean instances 
- I added test to cover scenario with 2 beans (without default and alternatives)
- I added test to cover scenario with more beans (from all tests covered by this PR) - 1 of them is default and 1 of them is alternative - to cover back-compatibility on ambiguous failure